perfdash/downloader.go

```diff
@@ -17,37 +17,36 @@ limitations under the License.
 package main
 
 import (
-	"k8s.io/kubernetes/pkg/util/sets"
-)
+	"encoding/json"
+	"fmt"
+	"net/http"
 
-// Histogram is a map from bucket to latency (e.g. "Perc90" -> 23.5)
-type Histogram map[string]float64
+	"k8s.io/kubernetes/test/e2e/perftype"
+)
 
-// APICallLatency represents the latency data for a (resource, verb) tuple
-type APICallLatency struct {
-	Latency  Histogram `json:"latency"`
-	Resource string    `json:"resource"`
-	Verb     string    `json:"verb"`
+// Downloader is the interface that gets a data from a predefined source.
+type Downloader interface {
+	getData() (TestToBuildData, error)
 }
 
-// LatencyData represents the latency data for a set of RESTful API calls
-type LatencyData struct {
-	APICalls []APICallLatency `json:"apicalls"`
+// BuildData contains jenkins project name and a map from build number to perf data
+type BuildData struct {
+	Builds  map[string][]perftype.DataItem `json:"builds"`
+	Project string                         `json:"project"`
 }
 
-// ResourceToHistogram is a map from resource names (e.g. "pods") to the relevant latency data
-type ResourceToHistogram map[string][]APICallLatency
-
-// TestToHistogram is a map from test name to ResourceToHistogram
-type TestToHistogram map[string]ResourceToHistogram
-
-// BuildLatencyData is a map from build number to latency data
-type BuildLatencyData map[string]ResourceToHistogram
-
-// TestToBuildData is a map from test name to BuildLatencyData
-type TestToBuildData map[string]BuildLatencyData
-
-// Downloader is the interface that gets a data from a predefined source.
-type Downloader interface {
-	getData() (TestToBuildData, sets.String, sets.String, error)
+// TestToBuildData is a map from test name to BuildData
+type TestToBuildData map[string]BuildData
+
+func (buildLatency *TestToBuildData) ServeHTTP(res http.ResponseWriter, req *http.Request) {
```

Comment:
https://golang.org/pkg/net/http/#Handler
This is the function we have to implement for http handler.